cmd/setup.go

```diff
@@ -200,6 +212,37 @@ func (u *User) newMFASession(logger *log.Logger) (*session.Session, error) {
 	return mfaSession, nil
 }
 
+// GetMFADevice gets the user's existing virtual MFA device and updates the
+// MFA serial in the profile field.
+func (u *User) GetMFADevice(logger *log.Logger) error {
+	logger.Println("Getting the existing MFA device...")
+
+	sess, err := u.newSession()
+	if err != nil {
+		return fmt.Errorf("unable to get new session: %w", err)
+	}
+	svc := iam.New(sess)
+
+	mfaDeviceInput := &iam.ListMFADevicesInput{
+		UserName: aws.String(u.Name),
+	}
+
+	mfaDeviceOutput, err := svc.ListMFADevices(mfaDeviceInput)
+	if err != nil {
+		return fmt.Errorf("unable to get MFA: %w", err)
+	}
+
+	if len(mfaDeviceOutput.MFADevices) == 0 {
+		return errors.New("no MFA devices registered")
+	}
+	mfaDevice := mfaDeviceOutput.MFADevices[0]
```

Comment:
Does this mean you're taking the first device? Is it possible to have more than 1? How do you support a user who wants to use an MFA device that isn't the first one?

Reply:
I didn't think the setup supported more than one. Now I'm curious.

Reply:
Yeah, there's room for several in the api return but in practice I've never ever seen more than one. I assume it's some kind of internal tooling where they can provision a new one while the old one is being deprovisioned.

Regardless, I think it's a difficult user experience to try to figure out which one the user wants to select given the output of the API. So it's probably safe to assume the first one is the right one and all other edge cases will not be handled.

Reply:
For safety I added in an error if more than one is returned.

Reply:
Thank you!